README.md

```diff
@@ -1,11 +1,242 @@
+
+## The `yield` Keyword
+
+The `yield` keyword, when used inside the body of a method, will allow you to call that method with a block and pass, or "yield", to that block. Think of the `yield` keyword as saying "stop executing the code in this method and instead execute the code in the block. Then, return to the code in the method." 
+
+Let's look at the following example: 
+
+```ruby
+def yielding
+  puts "the program is executing the code inside the method"
+  yield
+  puts "now we are back in the method"
+end
+```
+
+To call this method with a block, we use the following syntax:
+
+```ruby
+yielding {puts "the method has yielded to the block!"}
+```
+
+or:
+
+```ruby
+yielding do 
+	puts "the method has yielded to the block!"
+end
+```
+
+When we call `yielding` with the above block, we will output:
+
+```ruby
+"the program is executing the code inside the method"
+"the method has yielded to the block!"
+"now we are back in the method"
+```
+
+
+#### Yielding With Parameters
+
+The `yield` keyword can take parameters. In other words, if you use yield and give it an argument, it will pass that argument to the block and that data will become available to the code in the block. 
+
+For example:
+ 
+```ruby
+def yielding_with_arguments(num)
+  puts "the program is executing the code inside the method"
+  i = num
+  yield(i)
+  puts "now we are back in the method"
+end
+```
+
+To call this method, we invoke our method and give it an argument *and* we accompany that method call with a block that contains a placeholder for the yielded value, plus whatever code we want to enact with that yielded value:
+
+```ruby
+yielding_with_arguments(2) {|i| puts i * 2}
+```
+
+We call our method with an argument:
+
+```ruby
+yielding_with_arguments(2)
+```
+and a block:
+
+```ruby
+{ |i| puts i * 2 }
+```
+
+The `|i|` (placeholder variable in between pipes) is our placeholder for the yielded value. The `i * 2` is the code we actually want to enact with our yielded value. 
+
+So, the above method call will output:
+
+```ruby
+"the program is executing the code inside the method"
+4
+"now we are back in the method"
+```
+
+The syntax inside the block might look familiar—it is how we identify index items in a block when we call `#each` on an array and pass a block to that method call. 
+
+## Code Along: Building Our Own Method with Yield
+
+Let's revisit our earlier example of a call to the `#each` method that only `puts` out a greeting if the word we pass into the block starts with the letter `"T"`. 
+
+```ruby
+["Tim", "Tom", "Jim"].each do |name|
+	if name.starts_with?("T")
+		puts "Hi, #{name}"
+	end
+end
+```
+
+In this example, we'll be building our own method, `#hello_t`. 
+
+Open up `lib/hello.rb`. We'll be coding the body of the `#hello_t` method. 
+
+#### Step 1: Defining our method to accept an argument
+
+Our method needs to operate on an array so let's define it to take in an argument:
+
+```ruby
+# lib/hello.rb
+
+def hello_t(array)
```

Comment:
@SophieDeBenedetto Don't know if there's a reason to call this method `hello_t`. Why not `my_each`?